r/R/dplyr.R

```diff
@@ -619,9 +619,23 @@ collect.arrow_dplyr_query <- function(x, as_data_frame = TRUE, ...) {
     restore_dplyr_features(tab, x)
   }
 }
-collect.ArrowTabular <- as.data.frame.ArrowTabular
+collect.ArrowTabular <- function(x, as_data_frame = TRUE, ...) {
+  if (as_data_frame) as.data.frame(x, ...) else x
```

Comment:
Good catch.

I know the inline if...else is legal but I personally find it harder to read than if they're on separate lines:

```suggestion
  if (as_data_frame) {
    as.data.frame(x, ...) 
  } else {
    x
  }
```